### PR TITLE
refactor(editor): Migrate `ui.store` to use composition API (no-changelog)

### PR DESCRIPTION
**Pull request by RicardoE105**
## Summary

1. Convert `ui.store` to composition API.
2. Remove unused getters `isNodeView` and `areExpressionsDisabled`.
3. Remove `isPageLoading` and `mainPanelPosition` from the state as they are not used anymore.
4. ~~Clean up all the uses of `isAnyModalOpen` as we do not have that method in the store.~~
5. Fix type errors that came up with the change.
6. Fix type warnings except for the cycle dependency ones (will address in a later PR when we migrate all stores to use composition API)

@alexgrozav thinking a bit further:
1. Should we also organize the stores by logical concert?
2. Is this not too big of a store? We can split it into multiple ones. For instance, could we extract modals, banners, and nodes selected into its own stores?

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/stores/ui.store.ts [hunk 1]
```diff
@@ -87,545 +81,660 @@ try {
...
-				return `${linkUrl}?${searchParams.toString()}`;
-			};
-		},
-		headerHeight() {
-			const style = getComputedStyle(document.body);
-			return Number(style.getPropertyValue('--header-height'));
-		},
-		isAnyModalOpen(): boolean {
-			return this.modalStack.length > 0;
-		},
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:377]: This computed property seems to be used in multiple places. I remember using it for the pin data discovery tooltip. I don't think it's safe to remove.

File: packages/editor-ui/src/stores/ui.store.ts [hunk 2]
```diff
@@ -87,545 +81,660 @@ try {
...
+	data: string;
+	canDrop: boolean;
+	stickyPosition: null | XYPosition;
+};
+
+export const useUIStore = defineStore(STORES.UI, () => {
+	const activeActions = ref<string[]>([]);
+	const activeCredentialType = ref<string | null>(null);
+	const theme = ref<ThemeOption>(savedTheme);
+	const modals = ref<{ [key: string]: ModalState }>({
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:98]: nit: Would prefer to use `Record<string, ModalState>` for type

File: packages/editor-ui/src/stores/ui.store.ts [hunk 3]
```diff
@@ -87,545 +81,660 @@ try {
...
+		return null;
+	});
+
+	const isVersionsOpen = computed(() => {
+		return modals.value[VERSIONS_MODAL_KEY].open;
+	});
+
+	const isModalOpen = computed(() => {
+		return (name: ModalKey) => modals.value[name].open;
+	});
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:292]: Can we extract this to a computed property `modalOpenById` instead? (see comment about computed getters from the other PR)

It allows for caching support.

```
const modalOpenById = computed(() => {
    return Object.keys(modals.value).reduce((acc, modal) => {
        acc[modal.name] = modals.value[name].open;
        return acc;
    }, {});
});
```

File: packages/editor-ui/src/stores/ui.store.ts [hunk 4]
```diff
@@ -87,545 +81,660 @@ try {
...
+
+	const getFakeDoorByLocation = computed(() => {
+		return (location: IFakeDoorLocation) =>
+			fakeDoorFeatures.value.filter((fakeDoor) => fakeDoor.uiLocations.includes(location));
+	});
+
+	const getFakeDoorById = computed(() => {
+		return (id: string) => fakeDoorFeatures.value.find((fakeDoor) => fakeDoor.id.toString() === id);
+	});
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:317]: Same comment as above.

I know it's a lot to ask, but we'll need to do this for the other stores as well.

File: packages/editor-ui/src/stores/ui.store.ts [hunk 5]
```diff
@@ -87,545 +81,660 @@ try {
...
+		return ![
+			VIEWS.WORKFLOW.toString(),
+			VIEWS.NEW_WORKFLOW.toString(),
+			VIEWS.EXECUTION_DEBUG.toString(),
+		].includes(currentView.value);
+	});
+
+	const isActionActive = computed(() => {
+		return (action: string) => activeActions.value.includes(action);
+	});
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:329]: Same comment as above.

File: packages/editor-ui/src/stores/ui.store.ts [hunk 6]
```diff
@@ -87,545 +81,660 @@ try {
...
+			const searchParams = new URLSearchParams();
+
+			if (deploymentType === 'cloud' && hasPermission(['instanceOwner'])) {
+				const adminPanelHost = new URL(window.location.href).host.split('.').slice(1).join('.');
+				const { code } = await cloudPlanStore.getAutoLoginCode();
+				linkUrl = `https://${adminPanelHost}/login`;
+				searchParams.set('code', code);
+				searchParams.set('returnPath', '/account/change-plan');
+			} else {
+				linkUrl = N8N_PRICING_PAGE_URL;
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:367]: Same comment as above.

File: packages/editor-ui/src/stores/ui.store.ts [hunk 7]
```diff
@@ -87,545 +81,660 @@ try {
...
-				if (!seen.has(node.id)) {
-					seen.add(node.id);
+			return false;
+		});
+	});
+
+	const isNodeSelected = computed(() => {
+		return (nodeName: string): boolean => {
+			for (const [index] of selectedNodes.value.entries()) {
+				if (selectedNodes.value[index].name === nodeName) {
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:346]: Same comment as above.

File: packages/editor-ui/src/stores/ui.store.ts [hunk 8]
```diff
@@ -87,545 +81,660 @@ try {
...
+		bannerStack.value.push(name);
+	};
+
+	const clearBannerStack = () => {
+		bannerStack.value = [];
+	};
+
+	const getNotificationsForView = (view: VIEWS) => {
+		return pendingNotificationsForViews.value[view] ?? [];
+	};
```
Comment (alexgrozav) [inline packages/editor-ui/src/stores/ui.store.ts:653]: This function is used in only once place. Can we remove it and use `pendingNotificationsForViews` directly instead?

**Review by alexgrozav — CHANGES_REQUESTED** [review]
Looking good. Same comments about value caching like the previous PR. 

I completely agree about the splitting of this store! We could tackle that in a separate PR.

File: packages/editor-ui/src/stores/ui.store.ts (hunk 4 above)
Comment (RicardoE105) [reply packages/editor-ui/src/stores/ui.store.ts:317]: All good! I will do it for this one and for the other stores as well!

File: packages/editor-ui/src/stores/ui.store.ts (hunk 1 above)
Comment (RicardoE105) [reply packages/editor-ui/src/stores/ui.store.ts:377]: Huge overlooked. I must have had the wrong filters in the search and got misled into thinking it was not used anymore. Awesome catch. Thanks

File: packages/editor-ui/src/stores/ui.store.ts (hunk 3 above)
Comment (RicardoE105) [reply packages/editor-ui/src/stores/ui.store.ts:292]: Can't we just rename modals to modalsById and use it drectly?

`uiStore.modalsById[name].open`

Comment (RicardoE105) [reply packages/editor-ui/src/stores/ui.store.ts:292]: We can to the same with all the other computed properties bellow:

This:

```typescript

	const getModalActiveId = computed(() => {
		return (name: ModalKey) => modals.value[name].activeId;
	});
``` 

To:

```typescript
uiStore.modalsById[name].activeId

```

This:

```typescript

	const getModalMode = computed(() => {
		return (name: ModalKey) => modals.value[name].mode;
	});
``` 
 
To:

```typescript
uiStore.modalsById[name].mode
``` 

The only one that would be to verbose to do like that would be IsModalActive. For that we can maybe catch the length and only run the comparison every time.

Comment (alexgrozav) [reply packages/editor-ui/src/stores/ui.store.ts:292]: Good point, I agree. Let's use it directly! 

For isModalActive we can also have a computed property:

```
const isModalActiveById = computed(() => Object.keys(modalsById.value).reduce((acc, name) => {
  acc[name] = name === modalStack.value[0];
  return acc;
}, {}); 
```

**Review by alexgrozav — DISMISSED** [review]
Thank you for addressing all the feedback! 💪

**Review by alexgrozav — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.50.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.50.0)

**Pull request merged**